ui-episodedetails-compose/src/main/java/app/tivi/episodedetails/EpisodeDetails.kt

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.tivi.episodedetails
+
+import android.view.ViewGroup
+import androidx.annotation.DrawableRes
+import androidx.compose.Composable
+import androidx.compose.ambient
+import androidx.compose.state
+import androidx.compose.unaryPlus
+import androidx.lifecycle.LiveData
+import androidx.ui.core.Alignment
+import androidx.ui.core.Modifier
+import androidx.ui.core.Text
+import androidx.ui.core.dp
+import androidx.ui.core.setContent
+import androidx.ui.foundation.Clickable
+import androidx.ui.foundation.DrawImage
+import androidx.ui.foundation.VerticalScroller
+import androidx.ui.foundation.isSystemInDarkTheme
+import androidx.ui.layout.Arrangement
+import androidx.ui.layout.AspectRatio
+import androidx.ui.layout.Column
+import androidx.ui.layout.EdgeInsets
+import androidx.ui.layout.HeightSpacer
+import androidx.ui.layout.Padding
+import androidx.ui.layout.Row
+import androidx.ui.layout.Spacing
+import androidx.ui.layout.Stack
+import androidx.ui.material.MaterialTheme
+import androidx.ui.material.ripple.Ripple
+import androidx.ui.material.surface.Surface
+import androidx.ui.material.withOpacity
+import androidx.ui.res.stringResource
+import androidx.ui.text.style.TextOverflow
+import androidx.ui.tooling.preview.Preview
+import app.tivi.data.entities.Episode
+import app.tivi.data.entities.EpisodeWatchEntry
+import app.tivi.data.entities.PendingAction
+import app.tivi.data.entities.Season
+import app.tivi.episodedetails.compose.R
+import app.tivi.util.TiviDateFormatter
+
+/**
+ * This is a bit of hack. I can't make `ui-episodedetails` depend on any of the compose libraries,
+ * so I wrap `setContext` as my own function, which `ui-episodedetails` can use.
+ */
+fun composeEpisodeDetails(
```

Comment:
Shouldn't this be @Composable? You are emitting Composables below, I'm surprised this compiles/works

Reply:
Not sure? This is just a wrapper around `setContent()` which I assume doesn’t need to be `@Composable`